csv1.rb

```diff
@@ -0,0 +1,68 @@
+require 'csv'
+class CSVReader
+attr_accessor :name,:age,:department
+@@values = Array.new
+
+@@filename
+
+   def self.inherited(subclass)
+      #puts "#{subclass}.csv"
+      @@filename = "#{subclass}.csv"
+   end
+
+   def self.addval(name,age,department)
+	     csv =Array.new
+      	 @@values <<  [name, age, department]
+         Student1.instances.each do |employee|
+          csv << ["name", "age", "department"]
+          csv << [name, age, department]
+         end
+    end
+
+	
+	def self.find_by_name(name)
```

Comment:
With this any subclass of CSVReader class will get this find_by_name method. Which is not what we need.